test/controllers/users_controller_test.rb

```diff
@@ -2,49 +2,180 @@
 
 class UsersControllerTest < ActionDispatch::IntegrationTest
   before do
-    @user = create :user
-    @user_2 = create :user
-    sign_in @user
+    @user = create :user, role: 'admin'
+    @user_2 = create :user, role: 'cm', name: 'Billy Everyteen'
     @patient_1 = create :patient, name: 'Susan Everyteen'
     @patient_2 = create :patient, name: 'Yolo Goat'
+
+    sign_in @user
   end
 
-  describe 'create method' do
-    it 'should raise if user is not admin' do
+  describe 'authentication for restricted endpoints' do
+    %w[users_path new_user_path].each do |endpoint|
+      it "should redirect if user is not admin - #{endpoint}" do
+        User.role.values.reject { |role| role == :admin }.each do |role|
+          @user.update role: role
+          get send(endpoint.to_sym)
```

Comment:
I've never seen send() used before, is it being used as a rest call here?

Reply:
Send is being used here to call methods :)